catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/RequiredAttributesMetacardValidator.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.validation.ReportingMetacardValidator;
+import ddf.catalog.validation.ValidationException;
+import ddf.catalog.validation.impl.ValidationExceptionImpl;
+import ddf.catalog.validation.impl.report.MetacardValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.MetacardValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+public class RequiredAttributesMetacardValidator implements ReportingMetacardValidator {
+    private final String metacardTypeName;
+
+    private final Set<String> requiredAttributes;
+
+    public RequiredAttributesMetacardValidator(final String metacardTypeName,
+            final Set<String> requiredAttributes) {
+        this.metacardTypeName = metacardTypeName;
+        this.requiredAttributes = requiredAttributes;
+    }
+
+    @Override
+    public void validate(final Metacard metacard) throws ValidationException {
+        final MetacardValidationReport report = validateByReport(metacard);
+
+        final List<String> errors = report.getMetacardValidationViolations()
+                .stream()
+                .map(ValidationViolation::getMessage)
+                .collect(Collectors.toList());
+
+        final ValidationExceptionImpl exception = new ValidationExceptionImpl();
+        exception.setErrors(errors);
+        throw exception;
+    }
+
+    @Override
+    public MetacardValidationReport validateByReport(final Metacard metacard) {
+        final MetacardValidationReport report = new MetacardValidationReportImpl();
+
+        final MetacardType metacardType = metacard.getMetacardType();
+
+        if (metacardTypeName.equals(metacardType.getName())) {
```

Comment:
We may want to at least log the case when the metacard's metacard type doesn't match the validator' metacard type name.